src/hotspot/share/opto/superword.cpp

```diff
@@ -2488,7 +2488,6 @@ void SuperWord::output() {
       } else if (VectorNode::is_scalar_rotate(n)) {
         Node* in1 = low_adr->in(1);
         Node* in2 = p->at(0)->in(2);
-        assert(in2->bottom_type()->isa_int(), "Shift must always be an int value");
```

Comment:
Why you removed the assert? What other types can be seen here?
The following code (specifically `in2->get_int()`) will fail if it is not Integer type.

Reply:
The shift is either an int value or a vector of an int value. If it's a vector !in2->is_Con() will make the condition true and no need to check for get_int due to short-circuiting.

Reply:
Okay. IT should be fine because get_int() has own assert.